backends/graphics/atari/atari-graphics.h

```diff
@@ -0,0 +1,272 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef BACKENDS_GRAPHICS_ATARI_H
+#define BACKENDS_GRAPHICS_ATARI_H
+
+#include "backends/graphics/graphics.h"
+
+#include "common/array.h"
+#include "common/events.h"
+#include "common/rect.h"
+#include "graphics/surface.h"
+
+// maximum screen dimensions
+constexpr int SCREEN_WIDTH = 640;
+constexpr int SCREEN_HEIGHT = 480;
+
+// minimum overlay dimensions
+constexpr int OVERLAY_WIDTH = 320;
+constexpr int OVERLAY_HEIGHT = 240;
+
+// don't waste time and space with that enum operator nonsense...
```

Comment:
Umm, why? And why so grumpy?

Reply:
That was written after seeing solutions like [this one](https://stackoverflow.com/a/35672584/21009). It's ridiculous how much code one has to write if we want to have 100% correct enums as flags. But I can reword it if you like.

Reply:
That post makes no sense. We do use enums as flags in hundreds of places in ScummVM. What are you talking about?

Reply:
Can you give me an example? So you declare an enum with { 1<<0, 1<<1, ... } and then use  something like

enum Flags f = Flags::val1 | Flags::val2;

Maybe you already have such support in ScummVM, it's possible I overlooked it.

Reply:
I am missing your point. Is it a problem for you to refer to other enums?   Enums are just the named constants, you can do it as you like. An example:

https://github.com/scummvm/scummvm/blob/master/engines/saga2/idtypes.h#L382

e.g. 

```C++
enum {
	kSLDBoxXSize = 374,
	...
	kSLDBoxX     = (640 - kSLDBoxXSize) / 2,
```

Nothing prevents you from writing
```C++
enum foo {
  flag1 = 1 << 0,
  flag2 = 1 << 1
};

enum bar {
   flagCommon = foo::flag1 | foo::flag2
}
```

Or perhaps am I missing something?

Reply:
Yes and no, you are cheating here. :-) Your example gives an illusion that `foo` is a scope of some kind but it isn't. Define another `enum baz { flag1 ... };` there and you'll get a compilation error. And guess what, that's exactly what I would get because I have `ScaleMode::NONE` and `PendingScreenChange::NONE` in my code.

So I'm forced to use enum **class** for my enums and that brings all that crazy operator code (you can check it with your example, too, `enum class foo` no longer works). Hence the grumpy comment and hence the namespace workaround to keep my sanity.

Reply:
Well, make it simpler. kScaleModeNone, kPendingScreenChangeNone and you solve your problem

Reply:
Not a huge fan of C solutions but I brought this to myself. :) Changed so enums where the actual number matters is defined using k... constants and where it doesn't stick to enum classes. That way I don't have to change all my enums to their pre-C++11 variants.